website/cue/reference/components/sinks/base/clickhouse.cue

```diff
@@ -161,7 +161,12 @@ base: components: sinks: clickhouse: configuration: {
 	endpoint: {
 		description: "The endpoint of the Clickhouse server."
 		required:    true
-		type: string: examples: ["http://localhost:8123"]
+		type: string: examples: ["http://localhost:8123", "tcp://localhost:9000"]
+	}
+	use_native_proto: {
+		description: "If `true`, ClickHouse Native Protocol is used. Defaults to `false`, using `JSONEachRow` over HTTP."
+		required:    false
+		type: bool: default: false
 	}
```

Comment:
> Could derive this from the protocol? That is, if they specify tcp:// we use the native protocol, otherwise HTTP?

I think, yes, if we preserve the behavior "if scheme in the `endpoint` field is empty - use HTTP by default". If this is okay - we could do it. Just need to rework the implementation a little bit.

> Just checking: does the native protocol support the same tls options the HTTP one does?

I didn't test it, just checked the ClickHouse documentation regarding TLS configuration and didn't find differences between native and HTTP-based interfaces from the TLS configuration point of view.